File: Serene/Serene.Script/Administration/UserRole/UserRoleDialog.cs
namespace Serene.Administration
{
    using Serenity;
    using System;
    using System.Runtime.CompilerServices;

    [Imported]
    public class UserRoleDialog : TemplatedDialog<UserRoleDialogOptions>
    {
        public UserRoleDialog(UserRoleDialogOptions opt)
            : base(opt)
        {

        }
    }

    [Imported, Serializable]
    public class UserRoleDialogOptions
    {
        public int UserID { get; set; }
        public string Username { get; set; }
    }

    /* 
    This class has been ported to TypeScript. See UserRoleDialog.ts
    Code below is only a reference for those who want to use Saltaralle

    public class UserRoleDialog : TemplatedDialog<UserRoleDialogOptions>
    {
        private RoleCheckEditor permissions;
        
        public UserRoleDialog(UserRoleDialogOptions opt)
            : base(opt)
        {
            permissions = new RoleCheckEditor(this.ById("Roles"));

            UserRoleService.List(new UserRoleListRequest
            {
                UserID = options.UserID,
            }, response =>
            {
                permissions.Value = response.Entities.Select(x => x.ToString()).ToList();
            });
        }

        protected override DialogOptions GetDialogOptions()
        {
            var opt = base.GetDialogOptions();
            opt.Buttons = new List<DialogButton>
            {
                new DialogButton {
                    Text = Q.Text("Dialogs.OkButton"),
                    Click = delegate
                    {
                        UserRoleService.Update(new UserRoleUpdateRequest
                        {
                            UserID = options.UserID,
                            Roles = permissions.Value.Select(x => Int32.Parse(x, 10)).ToList(),
                        }, response => {
                            DialogClose();
                            Window.SetTimeout(delegate()
                            {
                                Q.NotifySuccess(Q.Text("Site.UserRoleDialog.SaveSuccess"));
                            }, 0);
                        });
                    },
                },
                new DialogButton {
                    Text = Q.Text("Dialogs.CancelButton"),
                    Click = DialogClose
                }
            };
            opt.Title = String.Format(Q.Text("Site.UserRoleDialog.DialogTitle"), options.Username);
            return opt;
        }

        protected override string GetTemplate()
        {
            return "<div id='~_Roles'></div>";
        }
    }
    */
}
 

File: Serene/Serene.Web/Modules/Administration/UserRole/UserRoleDialog.ts
namespace Serene.Administration {

    @Serenity.Decorators.registerClass()
    export class UserRoleDialog extends Serenity.TemplatedDialog<UserRoleDialogOptions> {

        private permissions: RoleCheckEditor;

        constructor(opt: UserRoleDialogOptions) {
            super(opt);

            this.permissions = new RoleCheckEditor(this.byId('Roles'));

            UserRoleService.List({
                UserID: this.options.userID
            }, response => {
                this.permissions.value = response.Entities.map(x => x.toString());
            });
        }

        protected getDialogOptions() {
            var opt = super.getDialogOptions();

            opt.buttons = [{
                text: Q.text('Dialogs.OkButton'),
                click: () => {
                    Q.serviceRequest('Administration/UserRole/Update', {
                        UserID: this.options.userID,
                        Roles: this.permissions.value.map(x => parseInt(x, 10))
                    }, response => {
                        this.dialogClose();
                        Q.notifySuccess(Q.text('Site.UserRoleDialog.SaveSuccess'));
                    });
                }
            }, {
                text: Q.text('Dialogs.CancelButton'),
                click: () => this.dialogClose()
            }];

            opt.title = Q.format(Q.text('Site.UserRoleDialog.DialogTitle'), this.options.username);
            return opt;
        }

        protected getTemplate() {
            return "<div id='~_Roles'></div>";
        }
    }

    export interface UserRoleDialogOptions {
        userID: number;
        username: string;
    }
}